src/coreclr/jit/llvmcodegen.cpp

```diff
@@ -847,25 +848,22 @@ void Llvm::buildLocalField(GenTreeLclFld* lclFld)
 
     // TODO-LLVM: if this is an only value type field, or at offset 0, we can optimize.
     Value* structAddrValue = getLocalAddr(lclNum);
-    Value* structAddrInt8Ptr = castIfNecessary(structAddrValue, Type::getInt8PtrTy(_llvmContext));
-    Value* fieldAddressValue = gepOrAddr(structAddrInt8Ptr, lclFld->GetLclOffs());
-    Value* fieldAddressTypedValue =
-        castIfNecessary(fieldAddressValue, getLlvmTypeForVarType(lclFld->TypeGet())->getPointerTo());
+    Value* fieldAddressValue = gepOrAddr(structAddrValue, lclFld->GetLclOffs());
 
-    mapGenTreeToValue(lclFld, _builder.CreateLoad(fieldAddressTypedValue));
+    mapGenTreeToValue(lclFld, _builder.CreateLoad(getLlvmTypeForVarType(lclFld->TypeGet()), fieldAddressValue));
 }
 
 void Llvm::buildLocalVarAddr(GenTreeLclVarCommon* lclAddr)
 {
     unsigned int lclNum = lclAddr->GetLclNum();
+    Value* localAddr = getLocalAddr(lclNum);
     if (lclAddr->OperIs(GT_LCL_FLD_ADDR))
     {
-        Value* bytePtr = castIfNecessary(getLocalAddr(lclNum), Type::getInt8PtrTy(_llvmContext));
-        mapGenTreeToValue(lclAddr, gepOrAddr(bytePtr, lclAddr->GetLclOffs()));
+        mapGenTreeToValue(lclAddr, gepOrAddr(localAddr, lclAddr->GetLclOffs()));
```

Comment:
I think we can delete the branch (`if (lclAddr->OperIs`...) now, since it will work for both `LCL_VAR_ADDR` and `LCL_FLD_ADDR`.

Reply:
Thanks, looks like it.